Application/components/Messaging/MessagingInterface.tsx

```diff
@@ -200,7 +198,7 @@ export function MessagingInterface({
   
       // IMPORTANT: Every time a new message is sent, we are also overwriting the chat history in the database.
       // We are doing this to ensure that the chat history is always up to date.
-      if (!privateChat) {
+      if (captureHistory) {
         try {
```

Comment:
- The error handling in asynchronous operations like fetching message history and sending messages to the server is minimal. Errors are logged to the console, but there's no mechanism to inform the user about the failure or attempt any retries. This can lead to a poor user experience if something goes wrong.